index.js

```diff
@@ -27,7 +28,8 @@ exports.handler = (event, context, callback) => {
 
 function process(s3Object, callback) {
     const configPath = path.resolve(__dirname, "config.json");
-    const processor  = new ImageProcessor(s3Object);
+    const fileSystem = new S3FileSystem();
+    const processor  = new ImageProcessor(fileSystem, s3Object);
```

Comment:
It's because I'm trying to follow the [Hexagonal architecture pattern approach](https://claudiajs.com/tutorials/designing-testable-lambdas.html), which really make it all nicely testable and simple. It's mostly because we keep all those layers (event layer and s3 file system layer) separate and allow them to meet in one place, which is the `ImageProcessor`.